main.js

```diff
@@ -0,0 +1,112 @@
+// Étape 2
+function getLatinCharacterList(letter) {
+    return letter.split(''); 
+  }
+  const latinToMorse = {
+	'A':'.-',
+	'B':'-...',
+	'C':'-.-.',
+	'D':'-..',
+	'E':'.',
+	'F':'..-.',
+	'G':'--.',
+	'H':'....',
+	'I':'..',
+	'J':'.---',
+	'K':'-.-',
+	'L':'.-..',
+	'M':'--',
+	'N':'-.',
+	'O':'---',
+	'P':'.--.',
+	'Q':'--.-',
+	'R':'.-.',
+	'S':'...',
+	'T':'-',
+	'U':'..-',
+	'V':'...-',
+	'W':'.--',
+	'X':'-..-',
+	'Y':'-.--',
+	'Z':'--..'
+}
+  const morseToLatin = {
+    '-': "T",
+    '--': "M",
+    '---': "O",
+    '--.': "G",
+    '--.-': "Q",
+    '--..': "Z",
+    '-.': "N",
+    '-.-': "K",
+    '-.--': "Y",
+    '-.-.': "C",
+    '-..': "D",
+    '-..-': "X",
+    '-...': "B",
+    '.': "E",
+    '.-': "A",
+    '.--': "W",
+    '.---': "J",
+    '.--.': "P",
+    '.-.': "R",
+    '.-..': "L",
+    '..': "I",
+    '..-': "U",
+    '..-.': "F",
+    '...': "S",
+    '...-': "V",
+    '....': "H"
+  }
+
+//   Étape 3
+  function translateLatinCharacter(letter) {
+    letter = letter.toUpperCase(); 
+    return latinToMorse[letter]; 
+  }
+  
+  function encode(text) {
+    let encodedText = '';
+  
+    for (let i = 0; i < text.length; i++) {
+      const letter = text[i];

```

Comment:
C'est juste : en JS, les strings peuvent se parcourir comme des listes. C'est généralement vrai dans les langages haut-niveau, mais restons vigilants : tout les langages de programmation n'offrent pas cette facilité.

Dans l'esprit de l'exercice, tu pouvais réutiliser `getLatinCharacterList` pour explicitement manipuler une liste à la place d'une string :
```suggestion
    letters = getLatinCharacterList(text);
    for (let i = 0; i < letters.length; i++) {
      const letter = letters[i];
```